EIPS/eip-5725.md

```diff
@@ -0,0 +1,189 @@
+     *   regardless of whether they are ready to claim
+     */
+    function vestingPayout(uint256 tokenId) external view returns (uint256 payout);
+
+    /**
+     * @notice Number of tokens for the NFT which can be claimed at the current timestamp
+     * @dev It is RECOMMENDED that this is calculated as the `vestedPayout()` value with the total
+     * amount of tokens claimed subtracted.
+     * @param tokenId The NFT token id
+     * @return payout The number of vested tokens for the NFT which have not been claimed yet
+     */
+    function claimablePayout(uint256 tokenId) external view returns (uint256 payout);
+
+    /**
+     * @notice The start and end timestamps for the vesting of the provided NFT
+     * MUST return the timestamp where no further increase in vestedPayout occurs for `vestingEnd`.
+     * @param tokenId The NFT token id
+     * @return vestingStart The beginning of the vesting as a unix timestamp
+     * @return vestingEnd The ending of the vesting as a unix timestamp
+     */
+    function vestingPeriod(uint256 tokenId) external view returns (uint256 vestingStart, uint256 vestingEnd);
+
+    /**
+     * @notice Token which is used to pay out the vesting claims
+     * @param tokenId The NFT token id
+     * @return token The token which is used to pay out the vesting claims
+     */
+    function payoutToken(uint256 tokenId) external view returns (address token);
+}
+```
+
+
+## Rationale
+<!-- The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. -->
+**vesting terms**
+- _vesting_: Tokens which are locked until a future date
+- _vested_: Tokens which have reached their unlock date. (The usage in this specification relates to the **total** vested tokens for a given Vesting NFT.)
+- _claimable_: Amount of tokens which can be claimed at the current `timestamp`. 
+- _timestamp_: The unix `timestamp` (seconds) representation of dates used for vesting. 
+
+**vesting functions**
+- `vestingPayout()` and `vestedPayout()` add up to the total number of tokens which can be claimed by the end of of the vesting schedule, which is also equal to `vestedPayoutAtTime()` with `type(uint256).max` as the `timestamp`.
+- `vestedPayout()` will provide the total amount of tokens which are eligible for release (including claimed tokens), while `claimablePayout()` provides the amount of tokens which can be claimed at the current `timestamp`.
+- `vestedPayoutAtTime()` provides functionality to iterate through the `vestingPeriod()` and provide a visual of the release curve. This allows for tools to iterate through a vesting schedule and create a visualization using on-chain data. It would be incredible to see integrations that create SVG visuals of vesting curves directly on-chain. 
```

Comment:
These bits read more like specification than rationale (since they're mostly describing behaviour and not the reasoning behind it.)